pageserver/src/tenant/storage_layer/inmemory_layer.rs

```diff
@@ -202,6 +214,85 @@ impl InMemoryLayer {
             Ok(ValueReconstructResult::Complete)
         }
     }
+
+    // Look up the keys in the provided keyspace and update
+    // the reconstruct state with whatever is found.
+    //
+    // If the key is cached, go no further than the cached Lsn.
+    pub(crate) async fn get_values_reconstruct_data(
+        &self,
+        keyspace: KeySpace,
+        end_lsn: Lsn,
+        reconstruct_state: &mut ValuesReconstructState,
+        ctx: &RequestContext,
+    ) -> Result<(), GetVectoredError> {
+        let ctx = RequestContextBuilder::extend(ctx)
+            .page_content_kind(PageContentKind::InMemoryLayer)
+            .build();
+
+        let inner = self.inner.read().await;
+
+        let reader = inner.file.block_cursor();
+
+        let mut completed_keys = HashSet::new();
+        let mut min_heap = BinaryHeap::new();
```

Comment:
Variable name of collections should state what they contain, not what they are.

Reply:
```suggestion
        let mut planned_block_reads = BinaryHeap::new();
```

Reply:
https://github.com/neondatabase/neon/pull/6576/commits/e46090dee699e0a4db125f82d6d4b71a42105378